src/services/implementation/ClientServiceImpl.java

```diff
@@ -0,0 +1,92 @@
+package services.implementation;
+
+import entities.Client;
+import enums.StatusEnum;
+import exceptions.services.EntityNotFoundException;
+import exceptions.services.IdNullPointerException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import services.interfaces.IClientInterface;
+
+public class ClientServiceImpl implements IClientInterface {
+  private List<Client> allClients;
+  private static ClientServiceImpl instance;
+  private Long id;
+
+  private ClientServiceImpl() {
+    this.allClients = new ArrayList<>();
+    this.id = 0L;
+  }
+
+  public static ClientServiceImpl getInstance() {
+    return Optional.ofNullable(instance).orElseGet(ClientServiceImpl::new);
+  }
+
+  @Override
+  public Optional<Client> insert(Client client) {
+    client.setId(this.id);
+    client.setCode("C" + this.allClients.size());
+    this.id++;
+    this.allClients.add(client);
+    return Optional.of(client);
+  }
+
+  @Override
+  public Optional<Client> findById(Long id) {
+    Optional.ofNullable(id).orElseThrow(IdNullPointerException::new);
+    Client client = null;
+    for (Client clientAux : this.allClients) {
+      if (clientAux.getId() == id && !(clientAux.getStatus().equals(StatusEnum.DELETED))) {
+        client = clientAux;
+        break;
+      }
+    }
+    Optional.ofNullable(client)
+        .orElseThrow(
+            () -> new EntityNotFoundException(Client.class.getClass().getSimpleName(), id));
+    return Optional.ofNullable(client);
+  }
+
+  @Override
+  public Optional<Client> delete(Long id) {
+    Optional.ofNullable(id).orElseThrow(IdNullPointerException::new);
+    Client client = this.findById(id).get();
+    client.setStatus(StatusEnum.DELETED);
+    return Optional.of(client);
+  }
+
+  @Override
+  public Optional<Client> update(Client object) {
+    Optional.ofNullable(object).orElseThrow(NullPointerException::new);
+    Optional.ofNullable(object.getId()).orElseThrow(IdNullPointerException::new);
+    Client client = this.findById(object.getId()).get();
+    client.setUpdatedAt(object.getUpdatedAt());
+    client.setCode(object.getCode());
+    client.setCreatedAt(object.getCreatedAt());
+    client.setAge(object.getAge());
+    client.setStatus(object.getStatus());
+    client.setCreatedBy(object.getCreatedBy());
+    client.setUpdatedby(object.getUpdatedby());
+    client.setName(object.getName());
+    client.setAddress(object.getAddress());
+    client.setPassword(object.getPassword());
+    client.setIdentification(object.getIdentification());
+    client.setLastName(object.getLastName());
+    client.setUsername(object.getUsername());
+    client.setUserRole(object.getUserRole());
+    client.setPhone(object.getPhone());
+    return Optional.of(client);
+  }
+
+  @Override
+  public Collection<Client> findAll() {
+    List<Client> clients =
```

Comment:
Just return the result of the filter, that variable is not doing anything

Reply:
solved